driver-core/src/main/com/mongodb/internal/connection/LoggingCommandEventSender.java

```diff
@@ -91,38 +113,41 @@ public void sendStartedEvent() {
         commandDocument = null;
     }
 
-    private String getTruncatedJsonCommand() {
-        StringWriter writer = new StringWriter();
-
-        try (BsonReader bsonReader = commandDocument.asBsonReader()) {
-            JsonWriter jsonWriter = new JsonWriter(writer,
-                    JsonWriterSettings.builder().outputMode(JsonMode.RELAXED).maxLength(MAX_COMMAND_DOCUMENT_LENGTH_TO_LOG).build());
-
-            jsonWriter.pipe(bsonReader);
-
-            if (jsonWriter.isTruncated()) {
-                writer.append(" ...");
-            }
-
-            return writer.toString();
-        }
-    }
-
     @Override
     public void sendFailedEvent(final Throwable t) {
         Throwable commandEventException = t;
         if (t instanceof MongoCommandException && redactionRequired) {
-            commandEventException = new MongoCommandException(new BsonDocument(), description.getServerAddress());
+            MongoCommandException originalCommandException = (MongoCommandException) t;
+            commandEventException = new MongoCommandException(new BsonDocument(), originalCommandException.getServerAddress());
+            commandEventException.setStackTrace(t.getStackTrace());
```

Comment:
This seems the right thing to do.

Reply:
does this logic preserve the error code, code name, and error labels? as discussed in the spec [here](https://github.com/mongodb/specifications/blob/master/source/command-logging-and-monitoring/command-logging-and-monitoring.rst#security)

Reply:
whoops, I meant to comment that one line up.

Reply:
No, it doesn't look like it.  Is there any way to add (unified) tests for that behavior?

Reply:
I don't think so, since we aren't requiring a specific error representation in the log messages and so drivers might just be asserting on a stringified error in the tests (this is what Rust is doing, since `tracing` doesn't allow us to attach non-primitive types to structured log messages). 

I tried to cover this in the test runner spec by saying 

> When failureIsRedacted is present and its value is true, the test runner MUST assert that a failure is present and that the failure has been redacted according to the rules defined for error redaction in the [command logging and monitoring specification](https://github.com/mongodb/specifications/blob/master/source/command-logging-and-monitoring/command-logging-and-monitoring.rst#security).

and then in the definition of redaction, the preservation of code/codename/labels is discussed. 

I wrote unit tests for the redaction functionality in Rust to confirm this. we could consider adding prose unit tests I suppose

Reply:
I'd like to defer this work to a separate PR, just to keep this one from getting even more complicated.

Reply:
Added JAVA-4843

Reply:
SGTM